### PR TITLE
Change ObjectSet's fields to properties

### PR DESCRIPTION
**Pull request by NextTurn**
Fixes an ancient issue.

CA1819 recommends to return a collection but it's gonna hurt the performance.

### CODE REVIEW

File: src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs [hunk 1]
```diff
@@ -16,17 +16,19 @@ public class ObjectSet
...
         /// </summary>
-        protected readonly int _minObjSize;
+        protected int MinObjSize { get; } = IntPtr.Size * 3;
 
         /// <summary>
         /// The collection of segments and associated objects.
         /// </summary>
-        protected HeapHashSegment[] _segments;
+#pragma warning disable CA1819 // Properties should not return arrays
+        protected HeapHashSegment[] Segments { get; }
```
Comment (leculver) [inline src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs:30]: Maybe we should just make all of these public and Segments an ImmutableArray?

Happy to merge as-is, but that would also satisfy the bug while making things a bit more open.

Comment (NextTurn) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs:30]: `ImmutableArray` implies the array elements won't change. Maybe something read-only like `ReadOnlyMemory`?

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs:30]: I don't read it that way.  Semantically ImmutableArray only enforces an immutable array of mutable objects.

In C++ turns, ImmutableArray is like `const Foo *` a "constant array of Foo objects" and not `Foo * const` which would be an "immutable array of mutable Foo objects" (and also not "const Foo * const" which is immutable/immutable).

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs]: Heh pretty sure I wrote that backwards.  It's been a really long day.

Comment (NextTurn) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs]: What???

Comment (NextTurn) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs]: We've made all of these read-only so it won't hurt to make them public.

Comment (leculver) [reply src/Microsoft.Diagnostics.Runtime/src/Common/ObjectSet.cs]: Makes sense, thanks!

**Pull request merged**